rewrite-java/src/main/java/org/openrewrite/java/AddOrUpdateAnnotationAttribute.java

```diff
@@ -250,7 +250,8 @@ public J.Annotation visitAnnotation(J.Annotation a, ExecutionContext ctx) {
                                     return it;
                                 }
                                 return ((J.Literal) it).withValue(newAttributeValue).withValueSource(newAttributeValue);
-                            } else {
+                            } else if (oldAttributeValue == null) {
```

Comment:
The test cases you added only seem to be situations where `oldAttributeValue` is not null, and you haven't adjusted any existing test expectations.
Which makes me wonder if we are actually capturing the behavior that's changed here